### PR TITLE
Don't crash during shutdown if we try creating metadata snapshots

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Windows Error Reporting is insisting it's surprisingly common for us to have a thread trying to create a compilation while Visual Studio is shutting down and we can't make COM calls anymore. This prevents that from happening. This is a tactical fix and we'll still let the rest of metadata snapshot system work normally.

*Review:* @dotnet/roslyn-ide

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
@dotnet-bot test vsi please.

**Comment by jasonmalinowski** [comment]
@dotnet-bot test vsi please

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs [hunk 1]
```diff
@@ -34,29 +33,39 @@ internal sealed partial class VisualStudioMetadataReferenceManager : IWorkspaceS
...
         private readonly ITemporaryStorageService _temporaryStorageService;
 
+        internal IVsXMLMemberIndexService XmlMemberIndexService { get; }
+
+        /// <summary>
+        /// The smart open scope service. This can be null during shutdown when using the service might crash. Any
+        /// use of this field or derived types should be synchronized with <see cref="_readerWriterLock"/> to ensure
+        /// you don't grab the field and then use it while shutdown continues.
+        /// </summary>
+        private IVsSmartOpenScope SmartOpenScopeServiceOpt { get; set; }
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs]: private read/write property?  Why not just a field?

**Comment by CyrusNajmabadi** [comment]
👍

**Pull request merged**